samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -26,12 +26,10 @@
 
 
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
-
-
   synchronized public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
     ZkConfig zkConfig = new ZkConfig(config);
     ZkClient zkClient = new ZkClient(zkConfig.getZkConnect(), zkConfig.getZkSessionTimeoutMs(), zkConfig.getZkConnectionTimeoutMs());
-    ZkUtils zkUtils = new ZkUtils(participantId, new ZkKeyBuilder(groupId), zkClient, zkConfig.getZkConnectionTimeoutMs());
```

Comment:
a) I would prefer to use processorId in the constructor parameter here, with more precise meaning of what's the participant in this particular context.
b) as long as the ZNode path is not affected by removing the input parameter here, it should be fine.

P.S. a side note to the ZkUtils class: usually, the list of entity names in the ZooKeeper methods are used to construct a znode path, which made me concerned about removing the participant parameter. Non-path related parameters usually occurs at the end.